warpscript/utapi/decodeCheckpoint.mc2

```diff
@@ -0,0 +1,14 @@
+<'
+syntax = "proto3";
+message Checkpoint {
+  int64 sizeD = 1;
+  int64 objD = 2;
+  int64 inB = 3;
+  int64 outB = 4;
+  map<string, int64> ops = 5;
+}
+'>
+PROTOC 'proto' STORE
+<%
+  HEX-> !$proto 'Checkpoint' PB->
+%>
```

Comment:
Maybe a simple unit test like you did for the snapshot decode/encode could be good to have, same for the other decode/encode macros

Reply:
Added tests for encode/decodeEvent
encodeCheckpoint is generalized to encodeRecord in the snapshot PR with tests